Update install recommendations in upgrade guides (#9086)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/core-upgrade/07-upgrading-to-v1.8.md

The [`run`](/reference/commands/run#the-`--empty`-flag) and [`build`](/reference/commands/build#the---empty-flag) commands now support the `--empty` flag for building schema-only dry runs. The `--empty` flag limits the refs and sources to zero rows. dbt will still execute the model SQL against the target data warehouse but will avoid expensive reads of input data. This validates dependencies and ensures your models will build properly.

> #### dbt-core and adapters are decoupled

Reviewer: ```suggestion
### dbt-core and adapters are decoupled
```

> Before dbt Core v1.8, dbt adapters directly depended on components of `dbt-core`, and `dbt-core` depended on the adapter for execution. This bidirectional dependency made it difficult to develop adapters independent of `dbt-core`. Beginning in v1.8, [`dbt-core` and adapters are decoupled](https://github.com/dbt-labs/dbt-adapters/discussions/87), making it easier for us to maintain and independently evolve adapters going forward.

Reviewer: ```suggestion
Before v1.8, dbt adapters directly depended on components of `dbt-core`, and `dbt-core` depended on the adapter for execution. This bidirectional dependency made it difficult to develop adapters independently. Starting in <Constant name="core" /> v1.8, [`dbt-core` and adapters are decoupled](https://github.com/dbt-labs/dbt-adapters/discussions/87), making it easier to maintain and evolve them independent of each other.
```

Reviewer 2: Should this be "Starting **in**"?

> For backwards compatibility, we have maintained the adapter packages' install-time dependencies on `dbt-core`. Even though they are technically not needed for adapter internals, users need both `dbt-core` and an adapter to execute dbt projects, and many users have been installing dbt with `pip install dbt-<adapter>` [since v1.0](/docs/dbt-versions/core-upgrade/Older%20versions/upgrading-to-v1.0#installation). That means `pip install dbt-snowflake` will continue to install the latest versions of both `dbt-core` and `dbt-snowflake`.

Reviewer: ```suggestion
For backward compatibility, adapter packages continue to depend on `dbt-core` at install time. Running a dbt project still requires both `dbt-core` and an adapter, and since v1.0, many users have installed both together using [`pip install dbt-<adapter>`](/docs/dbt-versions/core-upgrade/upgrading-to-v1.0#installation).

This behavior remains unchanged. For example, `pip install dbt-snowflake` installs the latest versions of both `dbt-core` and `dbt-snowflake`.
```
